dynawo/sources/Models/Modelica/Dynawo/Electrical/Sources/InjectorIDQ.mo

```diff
@@ -16,68 +16,67 @@ model InjectorIDQ "Injector controlled by d and q current components idPu and iq
+  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
   Modelica.Blocks.Interfaces.RealOutput PInjPuSn(start = -P0Pu * SystemBase.SnRef /SNom) "Injected active power in pu (base SNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 43}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.Blocks.Interfaces.RealOutput QInjPu(start = -Q0Pu) "Injected reactive power in pu (base SnRef)";
-  Modelica.Blocks.Interfaces.RealOutput PInjPu(start = -P0Pu) "Injected active power in pu (base SnRef)";
+  Modelica.Blocks.Interfaces.RealOutput QInjPuSn(start = -Q0Pu * SystemBase.SnRef /SNom) "Injected reactive power in pu (base SNom)" annotation(
+    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 5}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
+  Modelica.Blocks.Interfaces.RealOutput UPu(start = U0Pu) "Magnitude voltage at inverter terminal in pu (base UNom)" annotation(
+    Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, 81}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
   Modelica.ComplexBlocks.Interfaces.ComplexOutput uPu(re(start = u0Pu.re), im(start=u0Pu.im)) "Complex inverter terminal voltage, used as complex connector instead of terminal connector, terminal only used for physical connection, in pu (base UNom)" annotation(
     Placement(visible = true, transformation(extent = {{0, 0}, {0, 0}}, rotation = 0), iconTransformation(origin = {115, -33}, extent = {{-15, -15}, {15, 15}}, rotation = 0)));
 
-  // Internal variables:
-  Types.Angle UPhase(start = UPhase0) "Rotor angle: angle between machine rotor frame and port phasor frame";
-
-  parameter Types.VoltageModulePu U0Pu "Start value of voltage amplitude at injector terminal in pu (base UNom)";
-  parameter Types.Angle UPhase0 "Start value of voltage angle at injector terminal in rad";
-  parameter Types.ActivePowerPu P0Pu "Start value of active power in pu (base SnRef) (receptor convention)";
-  parameter Types.ReactivePowerPu Q0Pu "Start value of reactive power in pu (base SnRef) (receptor convention)";
-  parameter Types.ComplexVoltagePu u0Pu "Start value of complex voltage at injector terminal in pu (base UNom)";
-  parameter Types.ComplexApparentPowerPu s0Pu "Start value of apparent power at injector terminal in pu (base SnRef) (receptor convention)";
   parameter Types.ComplexCurrentPu i0Pu "Start value of complex current at injector terminal in pu (base UNom, SnRef) (receptor convention)";
   parameter Types.CurrentModulePu Id0Pu "Start value of idPu in pu (base SNom, UNom)";
   parameter Types.CurrentModulePu Iq0Pu "Start value of iqPu in pu (base SNom, UNom)";
```

Comment:
(base SNom, UNom) (generator convention)

Reply:
Is it really a current module?